Algorithm.CSharp/OptionUniverseFilterGreeksRegressionAlgorithm.cs

```diff
@@ -0,0 +1,148 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities.Option;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Regression algorithm demonstrating the option universe filter by greeks and other options data feature
+    /// </summary>
+    public class OptionUniverseFilterGreeksRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private const string UnderlyingTicker = "GOOG";
+        private Symbol _optionSymbol;
+        private bool _optionChainReceived;
+
+        public override void Initialize()
+        {
+            SetStartDate(2015, 12, 24);
+            SetEndDate(2015, 12, 24);
+            SetCash(100000);
+
+            AddEquity(UnderlyingTicker);
+            var option = AddOption(UnderlyingTicker);
+            _optionSymbol = option.Symbol;
+
+            SetOptionFilter(option);
+        }
+
+        protected virtual void SetOptionFilter(Option security)
+        {
+            // Contracts can be filtered by greeks, implied volatility, open interest:
+            security.SetFilter(u => u
+                .Delta(0.5m, 1.5m)
+                .Gamma(0.0001m, 0.0006m)
+                .Vega(0.01m, 1.5m)
+                .Theta(-2.0m, -0.5m)
+                .Rho(0.5m, 3.0m)
+                .ImpliedVolatility(1.0m, 3.0m)
+                .OpenInterest(100, 500));
+
+            // Note: there are also shortcuts for these filter methods:
+            /*
+            security.SetFilter(u => u
+                .D(0.5m, 1.5m)
+                .G(0.0001m, 0.0006m)
+                .V(0.01m, 1.5m)
+                .T(-2.0m, -0.5m)
+                .R(0.5m, 3.0m)
+                .IV(1.0m, 3.0m)
+                .OI(100, 500));
+            */
+        }
+
+        public override void OnData(Slice slice)
+        {
+            if (slice.OptionChains.TryGetValue(_optionSymbol, out var chain) && chain.Contracts.Count > 0)
+            {
+                Log($"[{Time}] :: Recieved option chain with {chain.Contracts.Count} contracts");
+                _optionChainReceived = true;
+            }
```

Comment:
Cool!
To be safe, think would be nice to assert the chain is respecting what we requested for the different attributes, same for python 🙏